internal/pkg/levee/propagation/propagation.go

```diff
@@ -248,6 +251,45 @@ func (prop *Propagation) visitOperands(n ssa.Node, maxInstrReached map[*ssa.Basi
 	}
 }
 
+func (prop *Propagation) visitSelect(sel *ssa.Select, maxInstrReached map[*ssa.BasicBlock]int, lastBlockVisited *ssa.BasicBlock) {
+	// Select returns a tuple whose first 2 elements are irrelevant for our
+	// analysis. The 3rd tuple element maps to the 1st Recv state, the 4th
+	// element maps to the 2nd state, and so on. Since Extract indices are
+	// 0-based, the Extract corresponding to the current Recv state
+	// therefore has an Index that is 1 greater than the current recvCount
+	// when that state is reached.
+	// See the ssa package code for more details.
+	recvCount := 0
+	extractIndex := map[*ssa.SelectState]int{}
+	for _, ss := range sel.States {
+		if ss.Dir == types.RecvOnly {
+			recvCount++
+			extractIndex[ss] = recvCount + 1
```

Comment:
The comment above feels like it's working to coerce 0-indexing into 1-indexing.  Consider flipping these instructions, i.e.

```
extractIndex[ss] = recvCount + 2
recvCount++
```
and using 0-indexing throughout.

The comment could be massaged at that point to something more like what you had prior.  "Select returns a tuple whose first two elements are unused for our analysis.  The remainder of the tuple corresponds one-to-one to channels in the Recv state."

Reply:
I considered doing that, but I'm not sure it's more readable. On the one hand, not having the "2" anywhere in the code feels confusing. On the other hand, writing `... = recvCount + 2` feels like it's incorrect because it's using an outdated value of `recvCount`.

That being said, I tried changing `recvCount` to `recvIndex` and I think that this naming change, combined with your proposal, is what makes the most sense.

Thank you for your valuable input!